src/components/GrampsjsMap.js

```diff
@@ -67,66 +70,94 @@ class GrampsjsMap extends LitElement {
     this.mapid = 'mapid'
     this.latitude = 0
     this.longitude = 0
+    this.year = -1
     this.latMin = 0
     this.latMax = 0
     this.longMin = 0
     this.longMax = 0
     this.layerSwitcher = false
+    this._currentLayer = 'OpenStreetMap'
   }
 
   firstUpdated() {
     const mapel = this.shadowRoot.getElementById(this.mapid)
     if (this.latMin === 0 && this.latMax === 0) {
-      this._map = new Map(mapel, {zoomControl: false}).setView(
+      this._map = new L.Map(mapel, {zoomControl: false}).setView(
         [this.latitude, this.longitude],
         this.zoom
       )
     } else {
-      this._map = new Map(mapel, {zoomControl: false}).fitBounds([
+      this._map = new L.Map(mapel, {zoomControl: false}).fitBounds([
         [this.latMin, this.longMin],
         [this.latMax, this.longMax],
       ])
     }
     const config = {...defaultConfig, ...window.grampsjsConfig}
-    const tileLayer = new TileLayer(config.leafletTileUrl, {
+    const tileLayer = new L.TileLayer(config.leafletTileUrl, {
       attribution: config.leafletTileAttribution,
       tileSize: config.leafletTileSize,
       zoomOffset: config.leafletZoomOffset,
       maxZoom: 19,
       zoomControl: false,
     })
     tileLayer.addTo(this._map)
-    this._map.addControl(control.zoom({position: 'bottomright'}))
+    this._gl = L.maplibreGL({
+      style: config.glStyle,
+    })
+    this._map.addControl(L.control.zoom({position: 'bottomright'}))
     if (this.locateControl) {
       this._map.addControl(
-        control.locate({position: 'bottomright', drawCircle: false})
+        L.control.locate({position: 'bottomright', drawCircle: false})
       )
     }
-    this._layercontrol = control.layers({OpenStreetMap: tileLayer}, null, {
-      position: 'bottomleft',
-    })
+    this._layercontrol = L.control.layers(
+      {
+        OpenHistoricalMap: this._gl,
+        OpenStreetMap: tileLayer,
+      },
+      null,
+      {
+        position: 'bottomleft',
+      }
+    )
     if (this.layerSwitcher) {
       this._map.addControl(this._layercontrol)
     }
+    this._map.on('baselayerchange', e => this._handleBaseLayerChange(e))
```

Comment:
> The OHM layer is initially not filtered

Add an event listener for the [`styledata`](https://maplibre.org/maplibre-gl-js/docs/API/type-aliases/MapEventType/#styledata) event. This will fire after MapLibre has downloaded enough of the stylesheet to manipulate it, but before the tiles are done loading:

```js
const mapLibreMap = this._gl.getMaplibreMap()
mapLibreMap.on('styledata', e => filterByDecimalYear(mapLibreMap, this.year))
```

Reply:
Yes, thanks!! I now put it into the `baselayerchanged` event handler, because apparently that `styledata` event listener has to be recreated each time the layer is changed, but now it works!